llvm/lib/Support/APInt.cpp

```diff
@@ -3094,3 +3094,19 @@ void llvm::LoadIntFromMemory(APInt &IntVal, const uint8_t *Src,
     memcpy(Dst + sizeof(uint64_t) - LoadBytes, Src, LoadBytes);
   }
 }
+
+APInt APIntOps::mulhs(const APInt &C1, const APInt &C2) {
+  assert(C1.getBitWidth() >= C2.getBitWidth());
```

Comment:
(style) Assertions should include a message string as well:
```
assert(C1.getBitWidth() >= C2.getBitWidth() && "BitWidth mismatch");
```

Reply:
The DivisionByConstantTest code allows different bitwidths - which isn't great - we should probably drop the DivisionByConstantTest diff and assert for `==` - we could then fix DivisionByConstantTest in a future patch?

Reply:
Yes I'd prefer that. If the bit widths don't match then it is no longer clear (to me) what these functions are supposed to do.

Reply:
Please raise a ticket for the DivisionByConstantTest. Thanks!